lib/assignment_pipeline.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Manages the current status of an assignment.
+class AssignmentPipeline
+  module AssignmentStatuses
+    IN_PROGRESS = 'in_progress'
+    READY_FOR_REVIEW = 'ready_for_review'
+    READY_FOR_MAINSPACE = 'ready_for_mainspace'
+  end
+
+  ASSIGNMENT_STATUS_PIPELINE = [
+    AssignmentStatuses::IN_PROGRESS,
+    AssignmentStatuses::READY_FOR_REVIEW,
+    AssignmentStatuses::READY_FOR_MAINSPACE
+  ]
+
+  module ReviewStatuses
+    PEER_REVIEWER_NEEDED = 'peer_reviewer_needed'
+    PEER_REVIEW_STARTED = 'peer_review_started'
+    PEER_REVIEW_COMPLETED = 'peer_review_completed'
+  end
+
+  REVIEW_STATUS_PIPELINE = [
+    ReviewStatuses::PEER_REVIEWER_NEEDED,
+    ReviewStatuses::PEER_REVIEW_STARTED,
+    ReviewStatuses::PEER_REVIEW_COMPLETED
+  ]
+
+  def initialize(flags:)
```

Comment:
Is the idea that `Assignment` will use this class by sending its own flags in, and then it will delegate things like `assignment_status` to this?